libvast/test/schema.cpp

```diff
@@ -298,6 +296,142 @@ TEST(parseable - complex types global) {
   CHECK(*e == *enum_t);
 }
 
+TEST(parseable - out of order definitions) {
+  using namespace std::string_view_literals;
+  auto str = R"__(
+    type baz = list<bar>
+    type bar = record{
+      x: foo
+    }
+    type foo = int
+  )__"sv;
+  schema sch;
+  CHECK(parsers::schema(str, sch));
+  auto baz = unbox(sch.find("baz"));
+  // clang-format off
+  auto ref = type{
```

Comment:
```suggestion
  auto expected = type{
```

In C++, `ref` usually means a reference in the sense of "pointer", which is why I would suggest another name.